provider/core.rb

```diff
@@ -91,7 +91,19 @@ def generate(output_folder, types, version_name)
     # rubocop:enable Metrics/PerceivedComplexity
 
     def copy_files(output_folder)
-      @config.files.copy.each do |target, source|
+      copy_file_list(output_folder, @config.files.copy)
+    end
+
+    def copy_common_files(output_folder, _version_name = nil)
```

Comment:
I'm not sure I understand the _version_name here?

Reply:
This is because `_version_name` is what our templates are using right now to swap on [google vs google-beta](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform/common~copy.yaml#L16) and because there is some [terrifying magic that injects local variables](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/compile/core.rb#L141) into the scope when templating. 

It looks like the reason we ended up adding a dependency on that particular variable naming was because it was referenced this way in the [original parse call](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/config.rb#L199) so I had to keep the naming convention the same.